mne/preprocessing/_csd.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2003-2010 Jürgen Kayser <rjk23@columbia.edu>
+# Copyright 2017 Federico Raimondo <federaimondo@gmail.com> and
+#                Denis A. Engemann <dengemann@gmail.com>
+#
+#
+# The original CSD Toolbox can be find at
+# http://psychophysiology.cpmc.columbia.edu/Software/CSDtoolbox/
+
+# Authors: Denis A. Engeman <denis.engemann@gmail.com>
+#          Alex Rockhill <aprockhill206@gmail.com>
+#
+# License: Relicensed under BSD (3-clause) and adapted with
+#          permission from authors of original GPL code
+
+import numpy as np
+
+from scipy.linalg import inv
+
+from mne import pick_types, pick_info
+from mne.utils import _validate_type
+from mne.io import BaseRaw
+from mne.epochs import BaseEpochs
+from mne.evoked import Evoked
+from mne.channels.interpolation import _calc_g, _calc_h
+
+
+def _prepare_G(G, lambda2):
+    G.flat[::len(G) + 1] += lambda2
+    # compute the CSD
+    Gi = inv(G)
+
+    TC = Gi.sum(0)
+    sgi = np.sum(TC)  # compute sum total
+
+    return Gi, TC, sgi
+
+
+def _compute_csd(data, G_precomputed, H, radius):
```

Comment:
I'll look into plotting, good call.

I'm not sure you what you mean by make radius a scalar optional parameter in the public function, I think it already is, wrapped in the optional parameter sphere.

Reply:
Yes, but if sphere is bound to anatomical measures then it will be out of a fine tuning.
I mean option rather for visualization than for calculation:  just like voltage of EEG signal is in 1e-6 V units but it is in uV on plots, then units for CSD are in V/m, but in uV/cm2 on plots.

Though some standard default configuration will be appropriate here, and some exotic scales like (uV/cm2 * 10) should be applied by hand on a ready plot.

Reply:
Not sure how to change units in topoplots in a way that makes sense. I looked into _handle_defaults and could make a special exception there but maybe @larsoner or @drammock or someone knows how to do that.

Reply:
We'll need to add the `V/m` unit to `fiff-constants` then use it here. Let me work on that first